grpc/src/main/java/com/linecorp/armeria/server/grpc/GrpcServiceBuilder.java

```diff
@@ -322,7 +322,112 @@ public GrpcServiceBuilder addService(String path, BindableService bindableServic
             return addService(path, interceptor, methodDescriptor);
         }
         registryBuilder.addService(path, bindableService.bindService(), methodDescriptor,
-                                   bindableService.getClass());
+                                   bindableService.getClass(), null);
+        return this;
+    }
+
+    /**
+     * Adds a gRPC {@link BindableService} with functions to this {@link GrpcServiceBuilder}.
+     * The functions will be applied in the specified order and before functions decorated with the service.
+     * In the following sample code, these functions will be applied
+     * in the order of FirstDecorator, SecondDecorator, ClassDecorator, and MethodDecorator.
```

Comment:
I feel explaining the insertion order of decorators is beyond the scope of this method.

How about adding it to https://armeria.dev/docs/server-grpc and leaving a link to it?

Reply:
It seems good!
I work on it.